mozci/task.py

```diff
@@ -358,6 +360,30 @@ def configuration(self):
         parts = config.split("-")
         return "-".join(parts[:-1] if parts[-1].isdigit() else parts)
 
+    @property
+    def overhead(self):
+        """Calculate the overhead of a task.
+
+        The methodology is simple: each task (action) has a start/end time.
+        Each group also has a start/end time. Take the earliest known group start
+        and latest known group end time, ensure the two falls somewhere in between
+        task start/end.
+
+        This definition of overhead does not take into account inter-group overhead
+        eg. restarting browser, teardown, etc.
+
+        Returns:
+            float: difference between task start/end and group start/end times.
+        """
+        data = run_query("test_task_overhead", Namespace(task_id=self.id))["data"].pop()
+        # Sanity check to ensure group start/end times are within task start/end.
+        assert data["task_min"] < data["group_min"]
+        assert data["task_max"] > data["group_max"]
```

Comment:
Do not crash on a problem:  We should be emitting problems as "warnings" to the logs and also declaring the overhead a zero (our best estimate).

Depanding on your logging library, something like:

    if not (data["task_min"] <= data["group_min"] <= data["group_max"] <= data["task_max"]):
        Log.warning(
            "task {{task.id}} has inconsistent group duration: {{data}}",  
            task=task,
            data=data
        )
        return 0

Reply:
Why shouldn't we crash on a problem?

I agree crashing on a problem is bad if you're writing an ETL, but mozci is not an ETL, it's a library. Some consumers of the library will want to ignore errors and keep going (like an ETL). Other consumers of the library will want to fail loudly so they can be notified immediately when things go awry. The nice thing about crashing is that it let's the consumers decide how to handle the situation.

Though assertions are trickier to catch than a proper `Exception`, so if we're going to do this validation it should be using something like `ValueError` and the fact `ValueError` can be raised should be well documented.

Reply:
@ahal Since you are the owner, I will leave the error mitigation patterns to you. In the meantime, my reasons are:

* **We have a reasonable mitigation** - The `assert` is catching logic bugs: raising exception when the data is incoherent. The definition of `overhead` is our "best estimate of overhead": In the event of incoherent data, a zero estimate is a legitimate return value.
* **Less work for callers** - For every raised exception, there is at least one `try/catch` needed to handle that error.  All callers must implement their own mitigation, possibly the same mitigation. I doubt anyone will be writing an exception handler for these asserts.
* **Callers have less context** - Callers are less likely to know a suitable mitigation:  It is likely the caller will let the error propagate up to terminate the program.
* **asserts are for debugging** - These specific asserts exist to crash in the event of bad data; to crash in the event of unanticipated data; which is useful only in debugging. My suggestion is to shunt the error to `stderr` in production, and consume that channel offline to guide further development.
* **this is a rare event** - I do not expect these asserts to be broken

Yes, letting an exception escape the callee allows the caller to decide mitigation, but I am not advocating suppression of all exceptions. In this case, we have a reasonable mitigation strategy that I doubt any caller will ever write a handler for; plus it remains visible to anyone monitoring `stderr`.

Of course, I can always be wrong.  I am sure I have promoted past `warning`s to `error`s so that a better strategy can be implemented by a caller: In which case I advocate a more detailed description so the caller has the context to make the best decision:

    if not (data["task_min"] <= data["group_min"] <= data["group_max"] <= data["task_max"]):
        Log.error(
            "task {{task.id}} has inconsistent group duration: {{data}}",  
            task=task,
            data=data
        )